include/deal.II/distributed/fully_distributed_tria.h

```diff
@@ -0,0 +1,406 @@
+     *
+     * Once the ConstructionData `construction_data` has been constructed, the
+     * triangulation `tria` can be created by calling
+     * `tria.reinit(construction_data);`.
+     *
+     * @note This triangulation supports: 1D/2D/3D, hanging nodes,
+     *       geometric multigrid, and periodicity.
+     *
+     * @note Currently no modifications of the triangulation is supported after
+     *       it has been created, i.e., this type of mesh does not support
+     *       any form of adaptivity, not even simple global refinements and
+     *       coarsenings.
+     *
+     * @note Currently only simple periodicity conditions (i.e. without offsets
+     *       and rotation matrices - see also the documentation of
+     *       GridTools::collect_periodic_faces)) are supported.
+     *
+     * @author Peter Munch, 2019
+     */
+    template <int dim, int spacedim = dim>
+    class Triangulation
+      : public parallel::DistributedTriangulationBase<dim, spacedim>
+    {
+    public:
+      using cell_iterator =
+        typename dealii::Triangulation<dim, spacedim>::cell_iterator;
+
+      using active_cell_iterator =
+        typename dealii::Triangulation<dim, spacedim>::active_cell_iterator;
+
+      using CellStatus =
+        typename dealii::Triangulation<dim, spacedim>::CellStatus;
+
+      /**
+       * Configuration flags for fully distributed Triangulations to be set in
+       * the constructor. Settings can be combined using bitwise OR.
+       */
+      enum Settings
+      {
+        /**
+         * Default settings, other options are disabled.
+         */
+        default_setting = 0x0,
+        /**
+         * This flags needs to be set to use the geometric multigrid
+         * functionality. This option requires additional computation and
+         * communication.
+         */
+        construct_multigrid_hierarchy = 0x1
+      };
+
+      /**
+       * Constructor.
+       *
+       * @param mpi_communicator denotes the MPI communicator to be used for the
+       *                         triangulation.
+       * @param settings See the description of the Settings enumerator.
+       */
+      explicit Triangulation(MPI_Comm       mpi_communicator,
+                             const Settings settings = default_setting);
+
+      /**
+       * Destructor.
+       */
+      virtual ~Triangulation() = default;
+
+      /**
+       * Create a triangulation from the provided ConstructionData.
+       *
+       * @note The namespace dealii::fullydistributed::Util contains functions
+       *       to create ConstructionData.
+       *
+       * @note This is the function to be used instead of
+       * Triangulation::create_triangulation() for some of the other
+       * triangulations of deal.II.
+       *
+       * @param construction_data The data needed for this process.
+       */
+      void
+      reinit(const ConstructionData<dim, spacedim> &construction_data);
```

Comment:
Why not just call this function `create_triangulation` as well? That would limit the need for learning new interfaces.